axum-extra/src/response/file_stream.rs

```diff
@@ -0,0 +1,212 @@
+                let reader = Cursor::new(file_content);
+
+                // response file stream
+                let stream = ReaderStream::new(reader);
+                let resp = FileStream::new(stream)
+                    .file_name("test")
+                    .content_size(size)
+                    .into_response();
+                resp
+            }),
+        );
+
+        // Simulating a GET request
+        let response = app
+            .oneshot(Request::builder().uri("/file").body(Body::empty())?)
+            .await?;
+
+        // Validate Response Status Code
+        assert_eq!(response.status(), StatusCode::OK);
+
+        // Validate Response Headers
+        assert_eq!(
+            response.headers().get("content-type").unwrap(),
+            "application/octet-stream"
+        );
+        assert_eq!(
+            response.headers().get("content-disposition").unwrap(),
+            "attachment; filename=\"test\""
+        );
+        assert_eq!(response.headers().get("content-length").unwrap(), "42");
+
+        // Validate Response Body
+        let body: &[u8] = &response.into_body().collect().await?.to_bytes();
+        assert_eq!(
+            std::str::from_utf8(body)?,
+            "Hello, this is the simulated file content!"
+        );
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn response_half_file() -> Result<(), Box<dyn std::error::Error>> {
+        let app = Router::new().route(
+            "/half_file",
+            get(move || async move {
+                let mut file = tokio::fs::File::open("CHANGELOG.md").await.unwrap();
+
+                // get file size
+                let file_size = file.metadata().await.unwrap().len();
+                
+                // seek to the middle of the file
+                let mid_position = file_size / 2;
+                file.seek(SeekFrom::Start(mid_position)).await.unwrap();
+
+                // response file stream
+                let stream = ReaderStream::new(file);
+                let resp = FileStream::new(stream)
+                    .file_name("CHANGELOG.md")
+                    .content_size(mid_position)
+                    .into_response();
+                resp
+            }),
+        );
+
+        // Simulating a GET request
+        let response = app
+            .oneshot(Request::builder().uri("/half_file").body(Body::empty()).unwrap())
+            .await
+            .unwrap();
+
+        // Validate Response Status Code
+        assert_eq!(response.status(), StatusCode::OK);
+
+        // Validate Response Headers
+        assert_eq!(
+            response.headers().get("content-type").unwrap(),
+            "application/octet-stream"
+        );
+        assert_eq!(
+            response.headers().get("content-disposition").unwrap(),
+            "attachment; filename=\"CHANGELOG.md\""
+        );
+        assert_eq!(response.headers().get("content-length").unwrap(), "8098");
```

Comment:
This will change each time we change the `CHANGELOG.md` file no?
It'd be better to add a specific file only for this test?

Reply:
Sorry for the writing error here, should have actively gotten the length once